src/clib/pio_darray.c

```diff
@@ -372,50 +397,65 @@ int pio_write_darray_nc(file_desc_t *file, io_desc_t *iodesc, const int vid,
     GPTLstop("PIO:write_darray_nc");
 #endif
 
     return ierr;
 }
 
-/** Write a set of one or more aggregated arrays to output file
- * @ingroup PIO_write_darray
+/** Write a set of one or more aggregated arrays to output file.
  *
- * This routine is used if aggregation is enabled, data is already on the
- * io-tasks
+ * This routine is used if aggregation is enabled, data is already on
+ * the io-tasks
  *
- * @param[in] file: a pointer to the open file descriptor for the file that will be written to
- * @param[in] nvars: the number of variables to be written with this decomposition
- * @param[in] vid: an array of the variable ids to be written
- * @param[in] iodesc_ndims: the number of dimensions explicitly in the iodesc
- * @param[in] basetype : the basic type of the minimal data unit
- * @param[in] gsize : array of the global dimensions of the field to be written
- * @param[in] maxregions : max number of blocks to be written from this iotask
- * @param[in] firstregion : pointer to the first element of a linked list of region descriptions.
- * @param[in] llen : length of the iobuffer on this task for a single field
- * @param[in] maxiobuflen : maximum llen participating
- * @param[in] num_aiotasks : actual number of iotasks participating
- * @param[in] IOBUF: the buffer to be written from this mpi task
- * @param[in] frame : the frame or record dimension for each of the nvars variables in IOBUF
+ * @param file a pointer to the open file descriptor for the file
+ * that will be written to
+ * @param nvars the number of variables to be written with this
+ * decomposition
+ * @param vid: an array of the variable ids to be written
+ * @param iodesc_ndims: the number of dimensions explicitly in the
+ * iodesc
+ * @param basetype the basic type of the minimal data unit
+ * @param gsize array of the global dimensions of the field to
+ * be written
+ * @param maxregions max number of blocks to be written from
+ * this iotask
+ * @param firstregion pointer to the first element of a linked
+ * list of region descriptions.
+ * @param llen length of the iobuffer on this task for a single
+ * field
+ * @param maxiobuflen maximum llen participating
+ * @param num_aiotasks actual number of iotasks participating
+ * @param IOBUF the buffer to be written from this mpi task
+ * @param frame the frame or record dimension for each of the nvars
+ * variables in IOBUF
+ *
+ * @return 0 for success, error code otherwise.
+ * @ingroup PIO_write_darray
  */
-int pio_write_darray_multi_nc(file_desc_t *file, const int nvars, const int vid[],
-			      const int iodesc_ndims, MPI_Datatype basetype, const PIO_Offset gsize[],
+int pio_write_darray_multi_nc(file_desc_t *file, const int nvars, const int *vid,
+			      const int iodesc_ndims, MPI_Datatype basetype, const PIO_Offset *gsize,
 			      const int maxregions, io_region *firstregion, const PIO_Offset llen,
 			      const int maxiobuflen, const int num_aiotasks,
-			      void *IOBUF, const int frame[])
```

Comment:
Is there a difference between const int frame[] and const int *frame?  Personal preference?